File: OpenGUI-UnityScript/Scripts/OGSlider.js
#pragma strict

@script AddComponentMenu ("OpenGUI/Slider")

enum SliderType {
	Horizontal,
	Vertical
}

class OGSlider extends OGWidget {
	var sliderType : SliderType;
	var sliderValue : float = 1.0;
	var min : float = 0.0;
	var max : float = 1.0;
		
	override function Draw ( x : float, y : float ) {	
		if ( sliderType == SliderType.Horizontal ) {
			sliderValue = GUI.HorizontalSlider ( Rect ( x, y, transform.localScale.x, transform.localScale.y ), sliderValue, min, max );
		} else {
			sliderValue = GUI.VerticalSlider ( Rect ( x, y, transform.localScale.x, transform.localScale.y ), sliderValue, max, min );
		}
	}
}

File: C#/Scripts/OGSlider.cs
using UnityEngine;
using System.Collections;

[AddComponentMenu ("OpenGUI/Slider")]
public class OGSlider : OGWidget {

	public enum SliderType {
		Horizontal,
		Vertical
	}

	public SliderType sliderType;
	public float sliderValue = 1.0f;
	public float min = 0.0f;
	public float max = 1.0f;
		
	public override void Draw (  float x ,   float y   ){	
		if ( sliderType == SliderType.Horizontal ) {
			sliderValue = GUI.HorizontalSlider ( new Rect( x, y, transform.localScale.x, transform.localScale.y ), sliderValue, min, max );
		} else {
			sliderValue = GUI.VerticalSlider ( new Rect( x, y, transform.localScale.x, transform.localScale.y ), sliderValue, max, min );
		}
	}
}
